frontend/app/services/cardService.ts

```diff
@@ -1,38 +1,54 @@
 import { CardData } from '../types/card';
+import Constants from 'expo-constants';
 
-// Constants
 export const CARDS_PER_PAGE = 20;
-const IDIOMS_BACKEND_URL =
-  process.env.IDIOMS_BACKEND_URL || 'http://localhost:8000/idioms/';
-
-/**
- * Fetches a paginated list of cards from the backend
- * @param page - The page number to fetch (1-based)
- * @param limit - Number of cards per page
- * @param search - Optional search query
- * @returns Promise with array of cards
- */
+
+export const getBackendUrl = (): string => {
+  const rawUrl = Constants.expoConfig?.extra?.IDIOMS_BACKEND_URL;
+
+  if (!rawUrl?.trim()) {
+    throw new Error('Missing IDIOMS_BACKEND_URL in app config');
+  }
+
+  return rawUrl.trim().replace(/\/?$/, '/');
+};
+
+const IDIOMS_BACKEND_URL = getBackendUrl();
+
+const handleApiError = async (response: Response) => {
+  const errorText = await response.text();
+  console.error(`HTTP Error ${response.status}:`, errorText);
+  throw new Error(
+    `Server error: ${response.status} - ${errorText.slice(0, 100)}`,
+  );
+};
+
 export const fetchCards = async (
   page: number,
   limit: number = CARDS_PER_PAGE,
   search?: string,
 ): Promise<CardData[]> => {
-  const url = new URL(IDIOMS_BACKEND_URL);
+  const url = new URL('idioms/', IDIOMS_BACKEND_URL);
   url.searchParams.append('page', page.toString());
   url.searchParams.append('limit', limit.toString());
-  if (search) {
-    console.log(
-      'Search functionality is not currently supported by the backend.',
-    );
-    url.searchParams.append('search', search.trim());
-  }
+  if (search) url.searchParams.append('search', search.trim());
 
-  const response = await fetch(url.toString());
+  try {
+    const response = await fetch(url.toString(), {
+      headers: {
+        Accept: 'application/json',
+        'ngrok-skip-browser-warning': '1',
```

Comment:
We don't want this in our code, we shouldn't check in ngrok config here, is there a way to start ngrok with this setting? Let's please remove it here